docs/cudf/source/developer_guide/ARCHITECTURE.md

```diff
@@ -0,0 +1,208 @@
+# cuDF Architecture
+
+The cuDF library is a GPU-accelerated, [Pandas-like](https://pandas.pydata.org/) DataFrame library.
+pandas APIs provide users a greate deal of power and flexibility, and we aim to match that.
+As a result, a key design challenge for cuDF is finding the simplest, most performant approaches to mimic pandas APIs.
+
+At a high level, cuDF is structured in three layers, each of which serves a distinct purpose in this regard:
+
+1. The Frame layer: The user-facing implementation of pandas-like data structures.
+2. The Column layer: The core internal data structures used to bridge the gap to our lower-level implementations.
+3. The Cython layer: The wrappers around the fast C++ `libcudf` library.
+
+In this document we will review each of these layers, their roles, and the requisite tradeoffs.
+
+
+## The Frame layer
+
+Broadly speaking, the `Frame` layer is composed of two types of objects: indexed tables and indexes.
```

Comment:
I would be fine avoiding this wording, if we could find alternative language to introduce these concepts. What would you suggest? I think the most important thing that needs to come across here is the idea of indexes as a columnar object distinct from other types of "tables". Neither DataFrames (outside of pandas) nor Tables (in the sense of a SQL-like tabular data store in a database) have that concept. It is is unique to pandas (and cuDF) and understanding that is crucial because it underpins a lot of these classes and data structures.

Reply:
Then maybe something along these lines:

> The `Frame` layer provides the `Series`, `DataFrame`, and various `Index` classes, which form the core user-facing data structures in cuDF, and offer similar functionality to their Pandas counterparts. It is composed of various classes connected by inheritance:

<insert tree here>

<later parts  of this section  here>